pkg/cache/informer_cache.go

```diff
@@ -152,6 +153,19 @@ func (ic *informerCache) GetInformer(ctx context.Context, obj client.Object) (In
 	return i.Informer, err
 }
 
+func (ic *informerCache) getInformerForKind(ctx context.Context, gvk schema.GroupVersionKind, obj runtime.Object) (bool, *internal.Cache, error) {
+	switch {
+	case ic.failOnUnknown:
```

Comment:
nit: now that this is a bool, seems like `if` would be clearer than `switch` but not a big deal